realm/realm-library/src/androidTest/kotlin/io/realm/RealmModelManagedSetTester.kt

```diff
@@ -0,0 +1,397 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm
+
+import io.realm.entities.AllTypes
+import io.realm.entities.PrimaryKeyAsString
+import io.realm.entities.SetContainerClass
+import io.realm.entities.SyncDog
+import io.realm.kotlin.createObject
+import io.realm.rule.BlockingLooperThread
+import java.lang.IllegalArgumentException
+import kotlin.reflect.KFunction1
+import kotlin.reflect.KFunction2
+import kotlin.reflect.KProperty1
+import kotlin.test.*
+
+/**
+ * Generic tester for Realm models types of managed sets.
+ */
+class RealmModelManagedSetTester<T : RealmModel>(
+        private val testerName: String,
+        private val mixedType: MixedType? = null,
+        private val setGetter: KFunction1<AllTypes, RealmSet<T>>,
+        private val setSetter: KFunction2<AllTypes, RealmSet<T>, Unit>,
+        private val managedSetGetter: KProperty1<SetContainerClass, RealmSet<T>>,
+        private val managedCollectionGetter: KProperty1<SetContainerClass, RealmList<T>>,
+        private val unmanagedInitializedSet: List<T?>,
+        private val unmanagedNotPresentValue: T,
+        private val toArrayManaged: ToArrayManaged<T>
+) : SetTester {
+
+    private lateinit var managedTester: ManagedSetTester<T>
+    private lateinit var config: RealmConfiguration
+    private lateinit var looperThread: BlockingLooperThread
+    private lateinit var realm: Realm
+    private lateinit var managedInitializedSet: MutableList<T>
+    private lateinit var managedNotPresentValue: T
+
+    private fun initAndAssertEmptySet(realm: Realm = this.realm): RealmSet<T> {
+        val allTypesObject = createAllTypesManagedContainerAndAssert(realm, "unmanaged")
+        assertNotNull(allTypesObject)
+        val set = setGetter.call(allTypesObject)
+        assertTrue(set.isEmpty())
+        return set
+    }
+
+    override fun toString(): String = when (mixedType) {
```

Comment:
This should not take into consideration `mixedType` since this tester should not be used for `Mixed` - unless you plan to do so...?